packages/ui/src/components/TimeInputV2/index.tsx

```diff
@@ -0,0 +1,469 @@
+      } else if (
+        type === 'm' &&
+        newValue.m &&
+        Number.parseInt(newValue.m, 10) >= 10
+      ) {
+        refSeconds.current?.focus()
+      }
+
+      if (type === 'h') {
+        if (isCompleteHour(timeFormat, newValue.h)) {
+          refMinutes.current?.focus()
+        }
+      }
+    }
+
+    setTime(newValue)
+    onChange?.(newValue)
+  }
+
+  // Increase time with arrow up
+  const handleIncrease = (type: 'h' | 'm' | 's') => {
+    const newTime = { ...time }
+    const currentValue = Number.parseInt(
+      time[type] && time[type].length > 0 ? time[type] : '0',
+      10,
+    )
+
+    if (type === 'h' && timeFormat === 24) {
+      newTime[type] = (currentValue === 23 ? 0 : currentValue + 1)
+        .toString()
+        .padStart(2, '0')
+    } else if (type === 'h' && timeFormat === 12) {
+      newTime[type] = (currentValue === 12 ? 1 : currentValue + 1)
+        .toString()
+        .padStart(2, '0')
+    } else {
+      newTime[type] = (currentValue === 59 ? 0 : currentValue + 1)
+        .toString()
+        .padStart(2, '0')
+    }
+    setTime(newTime)
+    onChange?.(newTime)
+  }
+
+  // Decrease time with arrow down
+  const handleDecrease = (type: 'h' | 'm' | 's') => {
+    const newTime = { ...time }
+    const currentValue = Number.parseInt(
+      time[type] && time[type].length > 0 ? time[type] : '0',
+      10,
+    )
+
+    if (type === 'h' && timeFormat === 24) {
+      newTime[type] = (currentValue === 0 ? 23 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    } else if (type === 'h' && timeFormat === 12) {
+      newTime[type] = (currentValue === 1 ? 12 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    } else {
+      newTime[type] = (currentValue === 0 ? 59 : currentValue - 1)
+        .toString()
+        .padStart(2, '0')
+    }
+    setTime(newTime)
+    onChange?.(newTime)
+  }
+
+  // Go to next input
+  const handleNext = (type: 'h' | 'm' | 's') => {
+    if (type === 'h') refMinutes.current?.focus()
+    if (type === 'm') refSeconds.current?.focus()
+    if (type === 's' && timeFormat === 12) refPeriod.current?.focus()
+  }
+
+  // Go to previous input
+  const handlePrevious = (type: 'h' | 'm' | 's') => {
+    if (type === 'm') refHours.current?.focus()
+    if (type === 's') refMinutes.current?.focus()
+  }
+
+  return (
+    <Stack gap={0.5} className={className} id={id} data-testid={dataTestId}>
+      <Stack direction="row" gap={1} alignItems="center">
+        <Text as="label" prominence="strong" sentiment="neutral" variant="body">
+          {label}
+        </Text>
+        {required ? <AsteriskIcon size={8} sentiment="danger" /> : null}
+        {labelDescription ? (
+          <Text as="label" variant="bodySmall">
+            {labelDescription}
+          </Text>
+        ) : null}
+        {labelDescription && typeof labelDescription !== 'string'
+          ? labelDescription
+          : null}
```

Comment:
this is a bug, we will display labelDescription twice

```suggestion
        {labelDescription ? (
            {labelDescription}
        ) : null}
```